azurerm/resource_arm_shared_image_version_test.go

```diff
@@ -61,6 +61,58 @@ func TestAccAzureRMSharedImageVersion_basic(t *testing.T) {
 		},
 	})
 }
+
+func TestAccAzureRMSharedImageVersion_storageAccountType(t *testing.T) {
+	resourceName := "azurerm_shared_image_version.test"
+
+	ri := tf.AccRandTimeInt()
+	resourceGroup := fmt.Sprintf("acctestRG-%d", ri)
+	userName := "testadmin"
+	password := "Password1234!"
+	hostName := fmt.Sprintf("tftestcustomimagesrc%d", ri)
+	sshPort := "22"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMSharedImageVersionDestroy,
+		Steps: []resource.TestStep{
+			{
+				// need to create a vm and then reference it in the image creation
+				Config:  testAccAzureRMSharedImageVersion_setup(ri, testLocation(), userName, password, hostName),
+				Destroy: false,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureVMExists("azurerm_virtual_machine.testsource", true),
+					testGeneralizeVMImage(resourceGroup, "testsource", userName, password, hostName, sshPort, testLocation()),
+				),
+			},
+			{
+				Config: testAccAzureRMSharedImageVersion_imageVersionStorageAccountType(ri, testLocation(), userName, password, hostName, "Standard_LRS"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSharedImageVersionExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "managed_image_id"),
+					resource.TestCheckResourceAttr(resourceName, "target_region.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "target_region.0.storage_account_type", "Standard_LRS"),
+				),
+			},
+			{
+				Config: testAccAzureRMSharedImageVersion_imageVersionStorageAccountType(ri, testLocation(), userName, password, hostName, "Standard_ZRS"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSharedImageVersionExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "managed_image_id"),
+					resource.TestCheckResourceAttr(resourceName, "target_region.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "target_region.0.storage_account_type", "Standard_ZRS"),
+				),
```

Comment:
running the test this fails because the Blob still exists - could we split remove the `Standard_LRS` section above so that this only provisions a ZFS account (and update the test name?)

Reply:
I can do this, however I ran the acceptance tests for the `azure_shared_image_version` against origin/master and get a failure due to the errors:

```
        Error: Code="TargetDiskBlobAlreadyExists" Message="Blob https://accsa191113120502781918.blob.core.windows.net/vhds/myosdisk1.vhd already exists. Please provide a different blob URI as target for disk 'myosdisk1'."
        
          on /var/folders/v8/cmph72_n7lvbpnbp2g5ssc0m0000gn/T/tf-test596992779/main.tf line 64:
          (source code not available)
```

Reply:
The issue has been that I didn't want to make changes that would affect other resources/data tests pass as part of this PR given I wasn't confident that I was running the acceptance tests in the correct way due to them failing on remote.

Happy to make the changes here though, but I would just double check to confirm that they are actually failing on master, or perhaps I have a local configuration issue?